README.md
ArtifactTracker
===============

Rift MMORPG addon. ArtifactTracker displays the direction and distance to artifacts in the game. Based on ResourceTracker.

PS. It's not done yet (not even something you could call a first release).


CHANGELOG.txt
1.0 - Initial release. Based on latest resource tracker (2013-07-15).

Below is the original Resource Tracker description. May not be accurate.

If you have Docker installed, it will place an icon in the Dock. If you do not
have Docker, it will place an icon somewhere in the topleft of the screen.

The icon is a silver diamond on a minimap style button. The button will
initially be gray, and can be moved around by dragging.

Once it is in place, RIGHT CLICK the button, and it will gain some color. At
this point, the configuration screen can be accessed by LEFT CLICKing the
button.

This will display the RT Config screen, which lists all Mining/Foraging node
types. Use the checkboxes to select which you are interested in seeing
notifications for. At the bottom of the config screen, is a dark red box,
showing an arrow, a resource and some text. Using the mousewheel on this red
section will change the size of the font used.

When the configuration screen is visible, there will also be another blue bar
shown, labelled 'Resource Tracker'. This is the anchor for where the alerts will
appear. This can be moved around by dragging the blue bar.

NOTE: The arrow indicator is the COMPASS DIRECTION of the node - not a
      direction relative to your heading.

The arrow will be green or red (red indicates that the node is significantly
above or below you)


lib/LibVersionCheck/UI.lua
local myWindow

local function L(x) return Translations.LibVersionCheck.L(x) end 

function SetAction(button, time)
	button:EventAttach(Event.UI.Button.Left.Press, function ()
		LibVersionCheckSettings.showAgainAt=time
		myWindow:SetVisible(false)
	end, "LeftDown", -2)
end

function LibVersionCheck.createAndShowUI()
    local context=UI.CreateContext("LibVersionCheck")

    myWindow=UI.CreateFrame("RiftWindow", "LibVersionCheckMainFrame", context)
    myWindow:SetTitle(L("Addons are outdated"));
    myWindow:SetPoint("CENTER", UIParent, "CENTER", 0, 0)
    
    myWindow:SetWidth(500)
    
    local h1=UI.CreateFrame("Text", "H1", myWindow); h1:SetFontSize(24); h1:SetText(L("Addon")); h1:SetPoint("TOPLEFT", myWindow, "TOPLEFT", 30, 70);
    local h2=UI.CreateFrame("Text", "H2", myWindow); h2:SetFontSize(24); h2:SetText(L("my Version")); h2:SetPoint("TOPLEFT", myWindow, "TOPLEFT", 150, 70);
    local h3=UI.CreateFrame("Text", "H3", myWindow); h3:SetFontSize(24); h3:SetText(L("newest Version")); h3:SetPoint("TOPLEFT", myWindow, "TOPLEFT", 300, 70);

    local tf;
    local bf;
    local row=1;

    for k, v in pairs(LibVersionCheckVersions) do
	if v.myVersion and v.newestVersion
	and ""..v.myVersion ~= "0"
	and ""..v.myVersion < ""..v.newestVersion then
    	    tf=UI.CreateFrame("Text", "T1"..row, myWindow); tf:SetFontSize(16); tf:SetText(k); tf:SetPoint("TOPLEFT", myWindow, "TOPLEFT", 30, row*20+100)
	    tf=UI.CreateFrame("Text", "T2"..row, myWindow); tf:SetFontSize(16); tf:SetText(""..v.myVersion); tf:SetPoint("TOPLEFT", myWindow, "TOPLEFT", 150, row*20+100)
	    tf=UI.CreateFrame("Text", "T3"..row, myWindow); tf:SetFontSize(16); tf:SetText(""..v.newestVersion); tf:SetPoint("TOPLEFT", myWindow, "TOPLEFT", 300, row*20+100)
	    row=row+1
	end
    end

    myWindow:SetWidth(500)
    -- myWindow:SetHeight(row*20+100)
    
    tf=UI.CreateFrame("Text", "showagain", myWindow); tf:SetText(L("show again : ")); tf:SetPoint("TOPLEFT", myWindow, "BOTTOMLEFT", 30, -80);
    
    bf=UI.CreateFrame("RiftButton", "sanextlogin", myWindow); bf:SetText(L("next login")); bf:SetPoint("TOPLEFT", myWindow, "BOTTOMLEFT", 160, -80); SetAction(bf, 0);
    bf=UI.CreateFrame("RiftButton", "satomorrow",  myWindow); bf:SetText(L("tomorrow"));   bf:SetPoint("TOPLEFT", myWindow, "BOTTOMLEFT", 290, -80); SetAction(bf, Inspect.Time.Server()+12*3600)
    bf=UI.CreateFrame("RiftButton", "sainaweek",   myWindow); bf:SetText(L("in a week"));  bf:SetPoint("TOPLEFT", myWindow, "BOTTOMLEFT",  30, -50); SetAction(bf, Inspect.Time.Server()+7*86400-43200)
    bf=UI.CreateFrame("RiftButton", "sainamonth",  myWindow); bf:SetText(L("in a month")); bf:SetPoint("TOPLEFT", myWindow, "BOTTOMLEFT", 160, -50); SetAction(bf, Inspect.Time.Server()+30*86400-43200)
    bf=UI.CreateFrame("RiftButton", "sanever",     myWindow); bf:SetText(L("never"));      bf:SetPoint("TOPLEFT", myWindow, "BOTTOMLEFT", 290, -50); SetAction(bf, Inspect.Time.Server()+3650*86400-43200)

    myWindow:SetVisible(true)
end


lib/LibVersionCheck/main.lua
local AddonData, privateTable = ...


if LibVersionCheck then
	return
end

LibVersionCheck={}


local function L(x) return Translations.LibVersionCheck.L(x) end 
local debugMode=false

-- Function to hand over to Command.Message.Send when we're not really
-- interested in whether the message arrives or not.

local function ignoreme()
end


-- Called when variables are fully loaded. To avoid problems if some
-- addon calls us before our variables are there, we save the version
-- info of that addon in Temp...., and copy it to the real table here.

local TempLibVersionCheckVersions
local function varsLoaded(handle, identifier)
	if not LibVersionCheckSpy then
		LibVersionCheckSpy={}
	end
	if not LibVersionCheckSettings then
		LibVersionCheckSettings={}
	end
	if not LibVersionCheckSettings.showAgainAt then
		LibVersionCheckSettings.showAgainAt=0
	end
	if not LibVersionCheckVersions then
		LibVersionCheckVersions = {}
		if TempLibVersionCheckVersions then
			for k, v in pairs(TempLibVersionCheckVersions) do
				LibVersionCheckVersions[k]=tempLibVersionCheckVersions[k]
			end
		end
	end
end

-- Called when a unit becomes available. If we haven't queried them before
-- in this session, we send a query to them. They will only react if they
-- have LibVersionCheck loaded as well, so we keep the messages to a
-- player without this addon minimal.

local player=nil
local didQuery={}
local function unitAvailable(handle, units)
	if player==nil then
		player=Inspect.Unit.Detail("player")
	end
	if player==nil or player.availability==nil or player.availability ~= "full" then
		player=nil
		return
	end
	local now=Inspect.Time.Server()
	local k,v
	for k,v in pairs(units) do
		if k ~= nil and type(k) == "string" and k ~= player.id and not didQuery[k] then
			local unit = Inspect.Unit.Detail(k)
			if unit and unit.availability
			and unit.availability=="full"
			and unit.player==true
			then
				didQuery[k]=now
				if debugMode then print("querying "..unit.name) end
				Command.Message.Send(unit.name, "LibVersionCheck", "query", ignoreme)
			end
		end
	end
end

-- Called when we get a message. This can be a query from some other
-- player, in which case we put his name and our addon list to our
-- "need to tell them" list, or it might be a response, which means
-- we save it to our list. We do not send all version info immediately,
-- instead, we add it to a list, and process one list entry every 1/10
-- sec, to keep the neccesary bandwith down.

-- When we get a new version number, we also save the name of who sent
-- it to us, so we can find out who spreads bogus numbers, should someone
-- have "funny" ideas.

local sendToList={}
local lastSendProcessed=0

local function gotMessage(handle, from, type, channel, identifier, data)
	if identifier ~= "LibVersionCheck" then return end

	if debugMode then print (data) end

	if data:len()>=5 and data:sub(1,5) == "query" then
		if debugMode then
			print("Got Query from "..from)
		end
		if not LibVersionCheckSettings.developer then
			for k, v in pairs(LibVersionCheckVersions) do
				if v.myVersion and v.myVersion ~= 0 and v.myVersion ~= "0" then
					table.insert(sendToList, { from, k })
				end
			end
		end
	end
	
	if data:len()>=7 and data:sub(1,7) == "version" then
		if debugMode then print("got version info: "..data.." from "..from) end
		local token
		local tnum=1
		local parms={}
		for token in string.gmatch(data, "[^%s]+") do
			parms[tnum]=token
			tnum=tnum+1
		end
		if tnum==4 then
			if not LibVersionCheckVersions[parms[2]] then
				LibVersionCheckVersions[parms[2]] = { myVersion="0" }
			end

			if not LibVersionCheckVersions[parms[2]].newestVersion
			or     LibVersionCheckVersions[parms[2]].newestVersion < parms[3] then
			       LibVersionCheckVersions[parms[2]].newestVersion = parms[3]
			       LibVersionCheckVersions[parms[2]].sender = from
			end
			if LibVersionCheckSettings.spymode then
				if not LibVersionCheckSpy then LibVersionCheckSpy={} end
				if not LibVersionCheckSpy[from] then LibVersionCheckSpy[from]={} end
				LibVersionCheckSpy[from][parms[2]]=parms[3]
				LibVersionCheckSpy[from][parms[2].."_date"]=Inspect.Time.Server()
			end
		end
	end
end


-- Send information about my version to everyone who has requested it.
-- We send our version number, NOT the newest version number we found,
-- to avoid propagating bogus version numbers we got from others.

local function systemUpdate(handle)

	if #sendToList<1 then return end

	local now=Inspect.Time.Real()
	if now < lastSendProcessed+0.1 then return end
	lastSendProcessed=now
	
	local toSend=sendToList[1]
	table.remove(sendToList, 1);
	if debugMode then
		print("Sending version "..LibVersionCheckVersions[toSend[2]].myVersion.." of "..toSend[2].." to "..toSend[1])
	end
	Command.Message.Send(toSend[1], "LibVersionCheck", "version "..toSend[2].." "..LibVersionCheckVersions[toSend[2]].myVersion, ignoreme)
end


local function allAddonsLoaded(handle)
	local anythingToShow=false
	for k, v in pairs(LibVersionCheckVersions) do
		if v.myVersion and v.newestVersion
		and ""..v.myVersion ~= "0"
		and ""..v.myVersion < ""..v.newestVersion then
			print(L("Addon ")..k..L(" has version ")..v.myVersion..L(" but ")..v.sender..L(" uses version ")..v.newestVersion..".")
			anythingToShow=true
		end
	end
	
	if anythingToShow and Inspect.Time.Server() > LibVersionCheckSettings.showAgainAt then
		Command.System.Watchdog.Quiet()
		LibVersionCheck.createAndShowUI()
	end
end


-- The command line interface.

local function slashHandler(h, args)
	local r = {}
	local numargs = 1
	local inquote = false
	local token, tmptoken
	for token in string.gmatch(args, "[^%s]+") do
		--print(token)
		if token:sub(1, 1) == "\"" then
			tmptoken=""
			token=token:sub(2) -- handle "abc" case
			inquote=true
			--print("start qoute token="..token)
		end
		if inquote then
			--print ("in quote, last char: "..token:sub(-1))
			if token:sub(-1) == "\"" then
				inquote=false
				token=token:sub(1, -2)
				token=tmptoken .. token
				--print ("combined string is "..token)
			else
				tmptoken=tmptoken .. token .. " "
				--print ("tmp token is "..token)
			end
		end
		if not inquote then
			r[numargs] = token
			numargs=numargs+1
		end
	end
	if numargs>1 then
		if r[1] == "list" then
			for k, v in pairs(LibVersionCheckVersions) do
				print(k);
				dump(LibVersionCheckVersions[k]);
			end
		elseif (r[1] == "query") and numargs>2 then
			print("querying "..r[2])
			Command.Message.Send(r[2], "LibVersionCheck", "query", ignoreme)
		elseif (r[1] == "debug") then
			debugMode=1
			print("libVersionCheck now debugging");
		elseif (r[1] == "reset") then
			didQuery={}
		elseif (r[1] == "spy" and numargs>2 and r[2] == "on") then
			LibVersionCheckSettings.spymode = true
			print("Spy mode enabled")
		elseif (r[1] == "spy") then
			LibVersionCheckSettings.spymode = false
			print("Spy mode disabled")
		elseif (r[1] == "developer" and numargs>2 and r[2] == "on") then
			LibVersionCheckSettings.developer = true
			print("Developer mode enabled")
		elseif (r[1] == "developer") then
			LibVersionCheckSettings.developer = false
			print("Developer mode disabled")
		else
			print(L("Usage") .. ": /LibVersionCheck list | debug | query <name>")
		end
	else
		print(L("Usage") .. ": /LibVersionCheck list | debug | query <name>")
	end
end

-- This is the only function other addons will ever need to know about.

function LibVersionCheck.register(addonName, addonVersion)
	local whereToStore
	if not LibVersionCheckVersions then
		TempLibVersionCheckVersions = {}
		whereToStore=TempLibVersionCheckVersions
	else
		whereToStore=LibVersionCheckVersions
	end
	if not whereToStore[addonName] then
		whereToStore[addonName]={}
	end
	whereToStore[addonName].myVersion=""..addonVersion
end



Command.Message.Accept(nil, "LibVersionCheck")
Command.Event.Attach(Event.Addon.SavedVariables.Load.End, varsLoaded,      "Event.Addon.SavedVariables.Load.End")
Command.Event.Attach(Event.Addon.Startup.End, 		  allAddonsLoaded, "Event.Addon.Startup.End")
Command.Event.Attach(Event.Unit.Availability.Full, 	  unitAvailable,   "Event.Unit.Availability.Full")
Command.Event.Attach(Event.Message.Receive, 	  	  gotMessage,      "Event.Message.Receive")
Command.Event.Attach(Event.System.Update.Begin,   	  systemUpdate,    "Event.System.Update.Begin")
Command.Event.Attach(Command.Slash.Register("LibVersionCheck"),
							  slashHandler,    "Command.Slash.Register")


lib/LibVersionCheck/Translations.lua
if not Translations then Translations = {} end
if not Translations.LibVersionCheck then Translations.LibVersionCheck = {} end

local translationTable = {
	["German"] = {
		["LibVersionCheck"]		   = "LibVersionCheck",
		["Addon "]			   = "Addon ",
		[" has version "]		   = " hat Version ",
		[" but "]			   = ", aber ",
		[" uses version "]		   = " verwendet Version ",
		
		["Addons are outdated"]		= "Addons sind veraltet",
		["Addon"]			= "Addon",
		["my Version"]			= "Meine Version",
		["newest Version"]		= "Neueste Version",
		["show again : "]		= "wieder zeigen : ",
		["next login"]			= "nächster Login",
		["tomorrow"]			= "morgen",
		["in a week"]			= "in 1 Woche",
		["in a month"]			= "in 1 Monat",
		["never"]			= "niemals",
	},
	["French"] = {
	},
	["Russian"] = {
	},
}

function Translations.LibVersionCheck.L(x)
	local lang=Inspect.System.Language()
	if  translationTable[lang]
	and translationTable[lang][x] then
		return translationTable[lang][x]
	elseif lang == "English"  then
		return x
	else
		if not translationTable[lang] then translationTable[lang]={} end
		translationTable[lang][x]=x
		print ("No translation yet for '" .. lang .. "'/'" .. x .. "'")
		return x
	end
end


ArtifactTracker_Data.lua
local addon, data = ...

data.ITEMS = {
  ["MINING"] = {
    { k=1,  name={["EN"]="Tin Lode"}, icon="Data\\UI\\item_icons\\ore11a.dds", skill=1 },
    { k=2,  name={["EN"]="Copper Vein"}, icon="Data\\UI\\item_icons\\ore03a.dds", skill=35 },
    { k=3,  name={["EN"]="Iron Vein"}, icon="Data\\UI\\item_icons\\ore03b.dds", skill=75 },
    { k=4,  name={["EN"]="Chromite Vein"}, icon="Data\\UI\\item_icons\\ore02a.dds", skill=115 },
    { k=5,  name={["EN"]="Cobalt Lode"}, icon="Data\\UI\\item_icons\\ore11b.dds", skill=145 },
    { k=6,  name={["EN"]="Silver Vein"}, icon="Data\\UI\\item_icons\\ore08.dds", skill=145 },
    { k=7,  name={["EN"]="Gold Vein"}, icon="Data\\UI\\item_icons\\ore06a.dds", skill=150 },
    { k=8,  name={["EN"]="Titanium Vein"}, icon="Data\\UI\\item_icons\\ore07.dds", skill=185 },
    { k=9,  name={["EN"]="Carmintium Deposit"}, icon="Data\\UI\\item_icons\\ore12.dds", skill=225 },
    { k=10, name={["EN"]="Platinum Deposit"}, icon="Data\\UI\\item_icons\\ore09.dds", skill=230 },
    { k=11, name={["EN"]="Orichalcum"}, icon="Data\\UI\\item_icons\\ore17.dds", skill=250 },
    { k=12, name={["EN"]="Karthite Vein"}, icon="Data\\UI\\item_icons\\ore13.dds" },
    { k=13, name={["EN"]="Gantimite Vein"}, icon="Data\\UI\\item_icons\\ore04b.dds" },
    { k=14, name={["EN"]="Rhenium Vein"}, icon="Data\\UI\\item_icons\\ore02.dds", skill=375 }
  },
  ["WOOD"] = {
    { k=1, name={["EN"]="Yew Log"}, icon="Data\\UI\\item_icons\\softwood.dds", skill=1 },
    { k=2, name={["EN"]="Ashwood Log"}, icon="Data\\UI\\item_icons\\wood5.dds", skill=35 },
    { k=3, name={["EN"]="Oak Log"}, icon="Data\\UI\\item_icons\\wood2.dds", skill=75 },
    { k=4, name={["EN"]="Mahogany Log"}, icon="Data\\UI\\item_icons\\plant43.dds", skill=115 },
    { k=5, name={["EN"]="Kingswood Log"}, icon="Data\\UI\\item_icons\\wood4.dds", skill=145 },
    { k=6, name={["EN"]="Runebirch Log"}, icon="Data\\UI\\item_icons\\runebirch_log_b.dds", skill=185 },
    { k=7, name={["EN"]="Sagebrush Log"}, icon="Data\\UI\\item_icons\\sagebrush.dds", skill=225 },
    { k=8, name={["EN"]="Shadethorn Branch"}, icon="Data\\UI\\item_icons\\wood5a.dds", skill=250 },
    { k=9, name={["EN"]="Linden Timber"}, icon="Data\\UI\\item_icons\\wood8.dds", skill=290 },
    { k=10, name={["EN"]="Elm Timber"}, icon="Data\\UI\\item_icons\\wood1.dds", skill=335 },
    { k=11, name={["EN"]="Madrosa Timber"}, icon="Data\\UI\\item_icons\\wood6a.dds", skill=375 }
  },
  ["PLANTS"] = {
    { k=1, name={["EN"]="Coastal Glory"}, icon="Data\\UI\\item_icons\\plant12.dds", skill=1 },
    { k=2, name={["EN"]="Grieveblossom"}, icon="Data\\UI\\item_icons\\plant48.dds", skill=1 },
    { k=3, name={["EN"]="Creeperbrush"}, icon="Data\\UI\\item_icons\\plant3.dds", skill=30 },
    { k=4, name={["EN"]="Krakenweed"}, icon="Data\\UI\\item_icons\\plant34.dds", skill=60 },
    { k=5, name={["EN"]="Razorbrush"}, icon="Data\\UI\\item_icons\\plant43.dds", skill=70 },
    { k=6, name={["EN"]="Duskglory"}, icon="Data\\UI\\item_icons\\plant2b.dds", skill=80 },
    { k=7, name={["EN"]="Golden Nettle"}, icon="Data\\UI\\item_icons\\plant31a.dds", skill=80 },
    { k=8, name={["EN"]="Wyvernspurr"}, icon="Data\\UI\\item_icons\\plant26.dds", skill=115 },
    { k=9, name={["EN"]="Tattertwist"}, icon="Data\\UI\\item_icons\\plant7c.dds", skill=140 },
    { k=10, name={["EN"]="Roc Orchid"}, icon="Data\\UI\\item_icons\\plant33b.dds", skill=145 },
    { k=11, name={["EN"]="Drakefoot"}, icon="Data\\UI\\item_icons\\plant26a.dds", skill=180 },
    { k=12, name={["EN"]="Bloodshade"}, icon="Data\\UI\\item_icons\\plant40.dds", skill=200 },
    { k=13, name={["EN"]="Basiliskweed"}, icon="Data\\UI\\item_icons\\plant30.dds", skill=225 },
    { k=14, name={["EN"]="Tempestflower"}, icon="Data\\UI\\item_icons\\plant25a.dds", skill=230 },
    { k=15, name={["EN"]="Twilight Bloom"}, icon="Data\\UI\\item_icons\\plant29.dds", skill=250 },
    { k=16, name={["EN"]="Chimera's Cloak"}, icon="Data\\UI\\item_icons\\plant10a.dds", skill=290 },
    { k=17, name={["EN"]="Frazzleweed"}, icon="Data\\UI\\item_icons\\plant36b.dds", skill=335 },
    { k=18, name={["EN"]="Lucidflower"}, icon="Data\\UI\\item_icons\\plant22.dds", skill=375 }
  },
  ["FISH"] = {
    { k=1, name={["EN"]="School of Fish"}, icon="Data\\UI\\item_icons\\fish_43.dds", skill=375 },
    { k=2, name={["EN"]="School of Clever Fish"}, icon="Data\\UI\\item_icons\\fish_38.dds", skill=375 },
    { k=3, name={["EN"]="School of Rare Fish"}, icon="Data\\UI\\item_icons\\fish_39_a.dds", skill=375 },
    { k=4, name={["EN"]="School of Strangely Mutated Fish"}, icon="Data\\UI\\item_icons\\akylios_balloon.dds", skill=375 },
    { k=5, name={["EN"]="Sunken Boat"}, icon="Data\\UI\\item_icons\\magical_component6.dds", skill=375 }
  }
}

data.translate = {
  ["RU"] = {
    ["MINING"] = {
      { k=1, n="Оловянная жила"},
      { k=2, n="Медная жила"},
      { k=3, n="Железная жила"},
      { k=4, n="Хромитовая жила"},
      { k=5, n="Залежи кобальта"},
      { k=6, n="Серебряная жила"},
      { k=7, n="Золотая жила"},
      { k=8, n="Титановая жила"},
      { k=9, n="Месторождение карминтия"},
      { k=10, n="Платиновая жила"},
      { k=11, n="Орихалковая жила"},
      { k=12, n="Каритовая жила"},
      { k=13, n="Гантимитовая жила"},
      { k=14, n="Рениевая жила"}
    },
    ["WOOD"] = {
      { k=1, n="Древесина тиса"},
      { k=2, n="Древесина ясеня"},
      { k=3, n="Дубовое полено"},
      { k=4, n="Бревно красного дерева"},
      { k=5, n="Полено королевского дерева"},
      { k=6, n="Полено рунной березы"},
      { k=7, n="Стебель полыни"},
      { k=8, n="Полено сумрачного боярышника"},
      { k=9, n="Брус липы"},
      { k=10, n="Древесина вяза"},
      { k=11, n="Древесина Мадрозы"}
    },
    ["PLANTS"] = {
      { k=1, n="Прибрежная слава"},
      { k=2, n="Горецвет"},
      { k=3, n="Ползучий кустарник"},
      { k=4, n="Трава кракена"},
      { k=5, n="Бритвенник"},
      { k=6, n="Сумроцвет"},
      { k=7, n="Золотистая крапива"},
      { k=8, n="Виверновые шпоры"},
      { k=9, n="Тряпочник"},
      { k=10, n="Орхидея рух"},
      { k=11, n="Драконник"},
      { k=12, n="Кровотень"},
      { k=13, n="Василисник"},
      { k=14, n="Бурецвет"},
      { k=15, n="Сумеречник"},
      { k=16, n="Плащ химеры"},
      { k=17, n="Одолень-трава"},
      { k=18, n="Ясноцвет"}
    },
    ["FISH"] = {
      { k=1, n="Косяк рыбы"},
      { k=2, n="Косяк хитрой рыбы"},
      { k=3, n="Косяк редкой рыбы"},
      { k=4, n="Косяк странно мутировавшей рыбы"},
      { k=5, n="Затонувшая лодка"}
    }
  },
  ["DE"] = {
    ["MINING"] = {
      { k=1,  n="Zinnader"},
      { k=2,  n="Kupferader"},
      { k=3,  n="Eisenader"},
      { k=4,  n="Chromitader"},
      { k=5,  n="Kobaltader"},
      { k=6,  n="Silberader"},
      { k=7,  n="Goldader"},
      { k=8,  n="Titanader"},
      { k=9,  n="Carmintium-Lager"},
      { k=10, n="Platin-Lager"},
      { k=11, n="Orichalcum"},
      { k=12, n="Tuthonyader"},
      { k=13, n="Gantimitader"},
      { k=14, n="Rheniumader"}
    },
    ["WOOD"] = {
      { k=1, n="Eibenstamm"},
      { k=2, n="Eschenholzstamm"},
      { k=3, n="Eichenstamm"},
      { k=4, n="Mahagoni-Stamm"},
      { k=5, n="Königsholz-Stamm"},
      { k=6, n="Runenbirken-Stamm"},
      { k=7, n="Salbeistamm"},
      { k=8, n="Schattendorn-Ast"},
      { k=9, n="Lindenholz"},
      { k=10, n="Ulmenholz"},
      { k=11, n="Madrosaholz"}
    },
    ["PLANTS"] = {
      { k=1, n="Küstenpracht"},
      { k=2, n="Trauerblüte"},
      { k=3, n="Kreuchstrauch"},
      { k=4, n="Krakenkraut"},
      { k=5, n="Klingenstrauch"},
      { k=6, n="Dämmerruhm"},
      { k=7, n="Goldnessel"},
      { k=8, n="Lindwurmstachel"},
      { k=9, n="Fledderflechte"},
      { k=10, n="Rokh-Orchidee"},
      { k=11, n="Drakenfuß"},
      { k=12, n="Blutschatten"},
      { k=13, n="Basiliskenkraut"},
      { k=14, n="Sturmblume"},
      { k=15, n="Zwielichtblüte"},
      { k=16, n="Chimeras Mantel"},
      { k=17, n="Fransenkraut"},
      { k=18, n="Leuchtblume"}
    },
    ["FISH"] = {
      { k=1, n="Fischschwarm"},
      { k=2, n="Schwarm kluger Fische"},
      { k=3, n="Schwarm seltener Fische"},
      { k=4, n="Schwarm Seltsam Mutierter Fische"},
      { k=5, n="Sunken Boat"}
    }
  },
  ["FR"] = {
    ["MINING"] = {
      { k=1,  n="Filon de fer-blanc"},
      { k=2,  n="Veine de cuivre"},
      { k=3,  n="Veine de fer"},
      { k=4,  n="Veine de Chromite"},
      { k=5,  n="Filon de Cobalt"},
      { k=6,  n="Filon d'argent"},
      { k=7,  n="Veine d'or"},
      { k=8,  n="Veine de titane"},
      { k=9,  n="Gisement de carmintium"},
      { k=10, n="Gisement de platine"},
      { k=11, n="Orichalque"},
      { k=12, n="Veine de tuthonie"},
      { k=13, n="Veine de gantimite"},
      { k=14, n="Veine en rhénium"}
    },
    ["WOOD"] = {
      { k=1, n="Rondin d'if"},
      { k=2, n="Bûche de frêne"},
      { k=3, n="Bûche de chêne"},
      { k=4, n="Bûche d'acajou"},
      { k=5, n="Bûche de bois royal"},
      { k=6, n="Bûche de rune-bouleau"},
      { k=7, n="Bûche d'armoise"},
      { k=8, n="Branche d'ombrépine"},
      { k=9, n="Branche de tilleul"},
      { k=10, n="Branche d'orme"},
      { k=11, n="Branche de madrosa"}
    },
    ["PLANTS"] = {
      { k=1, n="Gloire côtière"},
      { k=2, n="Fleur de chagrin"},
      { k=3, n="Ronce grimpante"},
      { k=4, n="Herbe à kraken"},
      { k=5, n="Ronce tranchante"},
      { k=6, n="Gloire du couchant"},
      { k=7, n="Ortie dorée"},
      { k=8, n="Ergot de wyverne"},
      { k=9, n="Tournehaillon"},
      { k=10, n="Orchiderokh"},
      { k=11, n="Patte-de-drake"},
      { k=12, n="Ombresang"},
      { k=13, n="Herbasilic"},
      { k=14, n="Fleur de la Tempête"},
      { k=15, n="Fleur du crépuscule"},
      { k=16, n="Cape de la Chimère"},
      { k=17, n="Herbe calcinée"},
      { k=18, n="Floralucida"}
    },
    ["FISH"] = {
      { k=1, n="Banc de poissons"},
      { k=2, n="Banc de poissons malins"},
      { k=3, n="Banc de poissons rares"},
      { k=4, n="Banc de poissons étrangement mutés"},
      { k=5, n="Épave de bateau"}
    }
  },
}

for lk, lv in pairs(data.translate) do
  for rk, rv in pairs(lv) do
    for nk, nv in pairs(rv) do
      for ik, iv in pairs(data.ITEMS[rk]) do
        if iv.k == nv.k then
          data.ITEMS[rk][ik].name[lk] = nv.n
          --print(string.format("data.ITEMS[%s][%s].name[%s] = %s", rk, ik, lk, nv.n))
          break
        end
      end
    end
  end
end

data.LOOKUP = {}

for rk, rv in pairs(data.ITEMS) do
  for ni, nv in pairs(rv) do
    data.LOOKUP[nv.name["DE"]] = {rk=rk, i=ni, k=nv.k}
    data.LOOKUP[nv.name["FR"]] = {rk=rk, i=ni, k=nv.k}
    data.LOOKUP[nv.name["EN"]] = {rk=rk, i=ni, k=nv.k}
    data.LOOKUP[nv.name["RU"]] = {rk=rk, i=ni, k=nv.k}
  end
end

if Inspect.System.Language() == "French" then
  data.SYSLANG = "FR"
elseif Inspect.System.Language() == "German" then
  data.SYSLANG = "DE"
elseif Inspect.System.Language() == "Russian" then
  data.SYSLANG = "RU"
else
  data.SYSLANG = "EN"
end
